integration-tests/tests/base_test.go

```diff
@@ -36,16 +38,39 @@ import (
 
 func init() {
 	c := &check.C{}
+
+	waitForRegularPartition(c)
+
+	disableAutopilot(c)
+
+	copySnapd(c)
+}
+
+func waitForRegularPartition(c *check.C) {
 	// Workaround for bug https://bugs.launchpad.net/snappy/+bug/1498293
 	// TODO remove once the bug is fixed
 	// originally added by elopio - 2015-09-30 to the rollback test, moved
 	// here by --fgimenez - 2015-10-15
 	wait.ForFunction(c, "regular", partition.Mode)
+}
 
+func disableAutopilot(c *check.C) {
 	cli.ExecCommand(c, "sudo", "systemctl", "stop", "snappy-autopilot.timer")
 	cli.ExecCommand(c, "sudo", "systemctl", "disable", "snappy-autopilot.timer")
 }
 
+func copySnapd(c *check.C) {
```

Comment:
> this (remounting rw) won't work on a squashfs system

We have a big problem then :) lots of tests rely on being able to modify files after remounting rw...

Regarding the use of `systemd-activate`, that was the previous approach (see `func (s *snapdTestSuite) SetUpTest(c *check.C)` in the diff of `integration-tests/tests/snapd_test.go`) and i wasn't getting the uid of the user, in the RemoteAddr I was getting `uid=;x.x.x.x`, maybe it can be launched differently so that it get the user related info?